File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        D = {}

        for char in s:
            if char not in D:
                D[char] = 1
            elif char in D:
                D[char] = D.get(char, 0) + 1

        for char in t:
            if char not in D:
                D[char] = 1
            elif char in D:
                D[char] = D.get(char, 0) - 1

        for value in D.values():
            if value != 0:
                return False

        return True

File: Data Structures & Algorithms/duplicate-integer/submission-2.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        Dictionary<int,int> D = new Dictionary<int,int>();

       foreach(int number in nums)
       {
        if(D.ContainsKey(number))
        {
            D[number] = D[number] + 1;
        }
        else{
            D[number] = 1;
        }
        if(D[number]>1)
        {
            return true;
        }
       }

       foreach(int value in D.Values){
        if (value > 1)
        {
            return true;
        }
       }

       return false;
    }
}